modules/host/persist.go

```diff
@@ -187,32 +208,27 @@ func (h *Host) load() error {
 			MinUploadBandwidthPrice   types.Currency `json:"minimumuploadbandwidthprice"`
 		}
 	}
-	err = h.dependencies.loadFile(persistMetadata, &compatPersistence, filepath.Join(h.persistDir, settingsFile))
+	err := h.dependencies.loadFile(persistMetadata, &compatPersistence, filepath.Join(h.persistDir, settingsFile))
 	if err != nil {
 		return err
 	}
 	// Load the compat values, but only if the compat values are non-zero and
 	// the real values are zero.
-	if compatPersistence.FinancialMetrics.PotentialStorageRevenue.Cmp(types.ZeroCurrency) > 0 && p.FinancialMetrics.PotentialStorageRevenue.Cmp(types.ZeroCurrency) == 0 {
+	if compatPersistence.FinancialMetrics.PotentialStorageRevenue.Cmp(types.ZeroCurrency) > 0 && p.FinancialMetrics.PotentialStorageRevenue.IsZero() {
```

Comment:
use `!IsZero` instead of `Cmp(types.ZeroCurrency) > 0`. Currency values can't be negative.